pyne/alara.py

```diff
@@ -774,3 +806,43 @@ def _output_flux(ve, tag_flux,output,start,stop,direction):
 
     output += "\n\n"
     return output
+
+def _get_subvoxel_array(mesh, cell_mats):
+    """
+    This function returns an array of subvoxels.
+    Parameters
+    ----------
+    mesh : PyNE Mesh object
+        The Mesh object for which the geometry is discretized.
+
+    return : subvoxel_array: structured array
+        A sorted, one dimensional array, each entry containing the following
+        fields:
+
+            :svid: int
+                The index of non-void subvoxel id
+            :idx: int
+                The idx of the voxel
+            :scid: int
+                The cell index of the cell in that voxel
+
+    """
+    cell_number_tag = mesh.mesh.getTagHandle('cell_number')
+    subvoxel_array = np.zeros(0, dtype=[(b'svid', np.int64),
+                                                      (b'idx', np.int64),
+                                                      (b'scid', np.int64)])
+    temp_subvoxel = np.zeros(1, dtype=[(b'svid', np.int64),
+                                                      (b'idx', np.int64),
+                                                      (b'scid', np.int64)])
+    # calculate the total number of non-void sub-voxel
+    non_void_sv_num = 0
+    for i, _, ve in mesh:
+        for c, cell in enumerate(cell_number_tag[ve]):
+            if cell > 0:
+                if len(cell_mats[cell].comp):
```

Comment:
This is really a test for void, right?  Perhaps we should add a method to the PyNE material object called `nonvoid()`?

Reply:
Yes, that's a test for void. I agree with you that we should add a method for PyNE material. But PyNE material is wrote in Cython, there are several files need to be changed if we add a method. So, I will add that in a separate PR.